presto-main/src/main/java/com/facebook/presto/sql/gen/JoinProbeCompiler.java

```diff
@@ -311,34 +330,16 @@ private static void generateGetChannelCountMethod(ClassDefinition classDefinitio
                 .retInt();
     }
 
-    private static void generateAppendToMethod(
-            ClassDefinition classDefinition,
-            CallSiteBinder callSiteBinder,
-            List<Type> types,
-            List<Integer> probeOutputChannels,
-            List<FieldDefinition> blockFields,
-            FieldDefinition positionField)
+    private static void generateGetOutputChannelsMethod(ClassDefinition classDefinition, FieldDefinition probeOutputChannelsField)
     {
-        Parameter pageBuilder = arg("pageBuilder", PageBuilder.class);
         MethodDefinition method = classDefinition.declareMethod(
                 a(PUBLIC),
-                "appendTo",
-                type(void.class),
-                pageBuilder);
-
+                "getOutputChannels",
+                type(int[].class));
         Variable thisVariable = method.getThis();
-        int pageBuilderOutputChannel = 0;
-        for (int outputChannel : probeOutputChannels) {
-            Type type = types.get(outputChannel);
-            method.getBody()
-                    .comment("%s.appendTo(block_%s, position, pageBuilder.getBlockBuilder(%s));", type.getClass(), outputChannel, pageBuilderOutputChannel)
-                    .append(constantType(callSiteBinder, type).invoke("appendTo", void.class,
```

Comment:
Chatted with @dain, I will do a benchmark comparing the performance between the generated class and the 'plain' class. If the benchmark result is promising, we could also deprecated `TwoChannelJoinProbe`. Somehow, the benchmark and the refactoring of the generated class may stay in my backlog until this patch is merged given that's a separated task.